python/pyspark/rdd.py

```diff
@@ -221,6 +227,21 @@ def context(self):
     def cache(self):
         """
         Persist this RDD with the default storage level (C{MEMORY_ONLY}).
+
+        :py:meth:`cache` can be used in a 'with' statement. The RDD will be automatically
+        unpersisted once the 'with' block is exited. Note however that any actions on the RDD
+        that require the RDD to be cached, should be invoked inside the 'with' block; otherwise,
+        caching will have no effect.
```

Comment:
Agreed, especially since this is technically a new Public API that we are potentially committing to for the life of the 2.x line.